staging/src/k8s.io/cloud-provider/pkg/controller/controller_utils.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	clientretry "k8s.io/client-go/util/retry"
+	taintutils "k8s.io/kubernetes/pkg/util/taints"
```

Comment:
I don't think we're allowed to have dependencies to k/k here so we may have to add `pkg/util/taints` to the list in https://github.com/kubernetes/kubernetes/issues/69585. What do you think?

Reply:
Sorry, forgot it
Good catch! :+1:

Reply:
BTW, so many file or function have a copy between `k8s.io/cloud-provider` and `k/k`. i want to commit a proposal like [https://github.com/kubernetes/community/pull/2354](https://github.com/kubernetes/community/pull/2354), so we have a clear target where the file need to move. WDYT?

Reply:
Good idea! So we do already have a KEP for this and I had previously proposed some changes here https://github.com/kubernetes/community/pull/2628. But there were some more changes recently so I think that should update that KEP with proposed new locations for these dependencies. Let me update that KEP and we can discuss locations in that PR?

Reply:
updated the KEP https://github.com/kubernetes/community/pull/2628, PTAL stewart :)